src/common/src/types/mod.rs

```diff
@@ -948,7 +958,7 @@ impl ScalarImpl {
                 }
                 let mut fields = Vec::with_capacity(s.len());
                 for (s, ty) in str[1..str.len() - 1].split(',').zip_eq_debug(s.types()) {
-                    fields.push(Some(Self::from_text(s.trim().as_bytes(), ty)?));
+                    fields.push(Self::from_pretty(s.trim(), ty)?);
```

Comment:
`from_pretty` is only used for internal testing, so we introduced `.` and `(empty)` for convenience. But `from_text` is used for parsing literals, which is user-facing. I don't think it should depend on `from_pretty` here. Maybe we can follow the [literal syntax for struct values](https://www.postgresql.org/docs/current/rowtypes.html#ROWTYPES-CONSTRUCTING). It uses `` for NULL and `""` for empty string. e.g.
```
<I,I>
(,1)
(1,)
(,)
```

Reply:
Btw, what is the difference between `ScalarImpl::from_literal` and `ScalarImpl::from_text`? Can we merge them now? cc @xiangjinwu

Reply:
Yes they should be merged.

Reply:
> But from_text is used for parsing literals, which is user-facing. I don't think it should depend on from_pretty here. Maybe we can follow the literal syntax for struct values. It uses `` for NULL and "" for empty string.

Oh yes, my bad.